itou/www/geiq_eligibility_views/forms.py

```diff
@@ -0,0 +1,100 @@
+from django import forms
+from django.urls import reverse
+
+from itou.eligibility.models.geiq import GEIQAdministrativeCriteria
+from itou.siaes.enums import SiaeKind
+
+
+class GEIQAdministrativeCriteriaForm(forms.Form):
+
+    RADIO_FIELDS = [
+        "de_inscrit_depuis_moins_de_12_mois",
+        "deld_12_24_mois",
+        "detld_24_mois",
+    ]
+
+    # Foldable optional radio fields
+    pole_emploi_related = forms.CharField(max_length=100, required=False)
+
+    def __init__(self, user, siae, administrative_criteria, **kwargs):
+        assert user, "No user defined"
+        assert siae and siae.kind == SiaeKind.GEIQ, "This form is only valid for GEIQ"
```

Comment:
Moyen fan des `assert` dans le code, ça nous aide pas vraiment plus qu'une exception `AttributeError: 'NoneType' object has no attribute`, ça aide pas plus l'utilisateur et c'est pas une exception qu'on à l'habitude d'attraper.

Reply:
C'est pas la première remarque que je me prends avec l'habitude des `asserts` en python (hérité d'un passé douteux).
Pour ma part, même si c'est chiant, je trouve pas mal que ça claque vite en cas d'incohérence.
Peut-être une frustration du typage faible ...
Mais au final, j'adopterais ce qui semble le plus idiomatique en Python.

Reply:
Je prends la remarque sur la forme parce que voir une `AssertionError` n'est pas forcément courant dans notre code, mais sur le fond je reste toujours convaincu du fail-fast avec les `assert`

Reply:
Vu qu'on se retrouvais avec 2 moyens pour faire la même chose je me suis dit que j'allais regarder ce que dit la doc à propos de `assert` et `AssertionError`, et comme souvent ça plis le game ;).
> The current code generator emits no code for an assert statement when optimization is requested at compile time.
https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement

On ne peux donc pas se fier au `assert` pour protéger son code car dépendant qu'une option _runtime_.